astropy/cosmology/_utils.py

```diff
@@ -81,3 +85,29 @@ def all_cls_vars(obj: object | type, /) -> dict[str, Any]:
     """Return all variables in the whole class hierarchy."""
     cls = obj if isinstance(obj, type) else obj.__class__
     return functools.reduce(operator.__or__, map(vars, cls.mro()[::-1]))
+
+
+def all_parameters(obj: object | type, /) -> dict[str, Field | Parameter]:
+    """Get all fields of a dataclass, including those not-yet finalized.
+
+    Parameters
+    ----------
+    obj : object | type
```

Comment:
But any `type` is also an `object`.

Reply:
Yes, but it's still an informative message for the user, which in this case is us maintainers for this private function. 
`obj` can be an instance of a class or the class itself.

Reply:
✅